dora/core/common/src/main/java/alluxio/network/protocol/RPCMessageEncoder.java

```diff
@@ -68,9 +68,17 @@ protected void encode(ChannelHandlerContext ctx, RPCMessage in, List<Object> out
 
     if (payload != null && bodyBytes > 0) {
       Object output = payload.getNettyOutput();
-      Preconditions.checkArgument(output instanceof ByteBuf || output instanceof FileRegion,
-          "The payload must be a ByteBuf or a FileRegion.");
-      out.add(output);
+      Preconditions.checkArgument(output instanceof ByteBuf
+              || output instanceof FileRegion || output instanceof List,
+          "The payload must be a ByteBuf or a FileRegion or a List<DataFileChannel>.");
+      if (output instanceof List) {
+        List<DataBuffer> dataFileChannels = (List<DataBuffer>) output;
```

Comment:
Is it possible to add more type checking here to assert the `output` must be a list of `List<DataBuffer>`?

Reply:
Now it must be `List<DataBuffer>`. If not, there must be something wrong.